### PR TITLE
SPY-1213: Dockerize!

### PR DESCRIPTION
**Pull request by zbelzer**
Adds a Dockerfile and instructions on how to build a base image with our custom spark build.

There might be some extra libs installed here and we might be able to slim it down with an alpine image instead. Todo, I suppose.

@jayceeb @jharveysmith @markhamstra

### CODE REVIEW

**Comment by zbelzer** [comment]
Any thoughts here guys? It's really just the addition of a Dockerfile. We release infrequently enough that we can push to ecs manually.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+RUN curl -sL --retry 3 \
+  "http://archive.apache.org/dist/hadoop/common/hadoop-$HADOOP_VERSION/hadoop-$HADOOP_VERSION.tar.gz" \
+  | gunzip \
+  | tar -x -C /usr/ \
+ && rm -rf $HADOOP_HOME/share/doc
+
+RUN ln -nsf $HADOOP_HOME/ /usr/lib/hadoop
+
+# SPARK
+ENV SPARK_VERSION 1.6.2
```
Comment (markhamstra) [inline Dockerfile]: Where is this being used?  I can see the need to use "1.6", "1.6.2" or "1.6.2-csd-7" depending on the context.

Comment (zbelzer) [inline Dockerfile]: This becomes an environment variable in the running worker, master, sparky and ingest environments. However, it mostly just amounts to substitution below.

**Comment by zbelzer** [comment]
This image becomes the "basic spark/hadoop software package" for more specific images like sparky, ingest, worker and master.

**Comment by jayceeb** [comment]
LGTM

**Comment by zbelzer** [comment]
Is this failure related to my changes?

**Pull request merged**